### ISSUE TITLE
Error: flag name is required if you already have existing configs

### ISSUE THREAD

**Issue by maxdd** [influxdata/influxdata-docker#474]
Hi,
I'm running a docker-compose with grafana on a new machine.
My only env variables are the following (which normally are sufficient to work)

```
environment:
      - DOCKER_INFLUXDB_INIT_MODE=setup
      - DOCKER_INFLUXDB_INIT_USERNAME=${INFLUXDB_USERNAME}
      - DOCKER_INFLUXDB_INIT_PASSWORD=${INFLUXDB_PASSWORD}
      - DOCKER_INFLUXDB_INIT_ORG=${INFLUXDB_ORG}
      - DOCKER_INFLUXDB_INIT_BUCKET=${INFLUXDB_BUCKET}
      - DOCKER_INFLUXDB_INIT_ADMIN_TOKEN=${INFLUXDB_TOKEN}
      - INFLUXDB_REPORTING_DISABLED=true
```

but i receive the error:

`Error: flag name is required if you already have existing configs`

Below the full log

```
influxdb2   | 2021-04-01T07:41:27.387733750Z    info    pinging influxd...      {"system": "docker"}
influxdb2   | ts=2021-04-01T07:41:28.377067Z lvl=info msg="Welcome to InfluxDB" log_id=0TF5vej0000 version=2.0.4 commit=4e7a59bb9a build_date=2021-02-08T17:47:02Z
influxdb2   | ts=2021-04-01T07:41:28.384552Z lvl=info msg="Resources opened" log_id=0TF5vej0000 service=bolt path=/var/lib/influxdb2/influxd.bolt
influxdb2   | ts=2021-04-01T07:41:28.386351Z lvl=info msg="Bringing up metadata migrations" log_id=0TF5vej0000 service=migrations migration_count=14
influxdb2   | ts=2021-04-01T07:41:28.543369Z lvl=info msg="Using data dir" log_id=0TF5vej0000 service=storage-engine service=store path=/var/lib/influxdb2/engine/data
influxdb2   | ts=2021-04-01T07:41:28.543547Z lvl=info msg="Compaction settings" log_id=0TF5vej0000 service=storage-engine service=store max_concurrent_compactions=4 throughput_bytes_per_second=50331648 throughput_bytes_per_second_burst=50331648
influxdb2   | ts=2021-04-01T07:41:28.543578Z lvl=info msg="Open store (start)" log_id=0TF5vej0000 service=storage-engine service=store op_name=tsdb_open op_event=start
influxdb2   | ts=2021-04-01T07:41:28.543666Z lvl=info msg="Open store (end)" log_id=0TF5vej0000 service=storage-engine service=store op_name=tsdb_open op_event=end op_elapsed=0.090ms
influxdb2   | ts=2021-04-01T07:41:28.543740Z lvl=info msg="Starting retention policy enforcement service" log_id=0TF5vej0000 service=retention check_interval=30m
influxdb2   | ts=2021-04-01T07:41:28.543806Z lvl=info msg="Starting precreation service" log_id=0TF5vej0000 service=shard-precreation check_interval=10m advance_period=30m
influxdb2   | ts=2021-04-01T07:41:28.543915Z lvl=info msg="Starting query controller" log_id=0TF5vej0000 service=storage-reads concurrency_quota=10 initial_memory_bytes_quota_per_query=9223372036854775807 memory_bytes_quota_per_query=9223372036854775807 max_memory_bytes=0 queue_size=10
influxdb2   | ts=2021-04-01T07:41:28.545182Z lvl=info msg="Configuring InfluxQL statement executor (zeros indicate unlimited)." log_id=0TF5vej0000 max_select_point=0 max_select_series=0 max_select_buckets=0
influxdb2   | ts=2021-04-01T07:41:28.846783Z lvl=info msg=Starting log_id=0TF5vej0000 service=telemetry interval=8h
influxdb2   | ts=2021-04-01T07:41:28.846848Z lvl=info msg=Listening log_id=0TF5vej0000 transport=http addr=:9999 port=9999
influxdb2   | 2021-04-01T07:41:29.405788062Z    info    pinging influxd...      {"system": "docker"}
influxdb2   | 2021-04-01T07:41:29.446345062Z    info    got response from influxd, proceeding   {"system": "docker"}
influxdb2   | Error: flag name is required if you already have existing configs
influxdb2   | See 'influx setup -h' for help
influxdb2   | 2021-04-01T07:41:29.476713112Z    warn    cleaning bolt and engine files to prevent conflicts on retry    {"system": "docker", "bolt_path": "/var/lib/influxdb2/influxd.bolt", "engine_path": "/var/lib/influxdb2/engine"}
influxdb2 exited with code 1
```

**Comment by maxdd** [comment]
I think in entrypoint.sh there should be `--name` as argument passed to influx setup.

`influx setup --force --username test--password test  --org test --bucket test --token test --name test`

In fact via command line it works as expected

**Comment by danxmoran** [comment]
@maxdd we can definitely add another env var to help here.

Can you help me understand more about how you got into this situation? Did you have a `configs` file that you were using outside of Docker, which you're now mounting into the container? Or did you hit this error by rebooting your container? The entrypoint logic is supposed to skip any setup logic if the DB's KV store is already present on disk, I want to make sure that isn't broken.

**Comment by maxdd** [comment]
Mhmm honestly i thought the behaviour was reproducible out of the box with the most recent container but it turned out its not happening if i start from scratch.
I am pretty sure my influxdb local volume was empty and i had no configs whatsoever.
What i was trying to do was to perform a setup mode but for a reason i dont know now it's not happening anymore

**Comment by seyoung720** [comment]
I'm running into this same issue running Influxdb 2.0 in a container. Where do I look for the config on my local filesystem and how do I resolve this? Many thanks.

**Comment by seyoung720** [comment]
Actually located the config in ~/.influxdbv2 is anyone runs in the same issue.

**Comment by LQss11** [comment]
TL;DR
This issue has to do something with influx-configs  in /etc/influxdb2 inside the container
I am getting the same issue too, and I have noticed the same behaviour happen when I try to bind a script  to /docker-entrypoint-initdb.d  so it could run and it's causing the url parameter in /etc/influxdb2/influx-configs  to be wrong -> "9999" instead "8086" I don't know where it's getting that port value but there must be something wrong with env variable or something.
Let me know if you found any solution for that.

**Comment by LQss11** [comment]
Anyways I found out my issue was because I was using a script inside the influx container that uses the CRLF EOL conversion that was getting executed on the first launch, turned out that I was cloning my project with autocrlf true, you can check more in this [StackOverflow reply](https://stackoverflow.com/questions/1967370/git-replacing-lf-with-crlf)
